File: algorithms/leetcode/442.find-all-duplicates-in-an-array/findDuplicates.cpp
// https://leetcode.com/problems/find-all-duplicates-in-an-array/
class Solution {
public:
  vector<int> findDuplicates(vector<int> &nums) {
    vector<int> result;
    for (int i = 0; i < nums.size(); i++) {
      int n = nums[i];
      if (n < 0) n *= -1;
      int flagIndex = n - 1;

      if (nums[flagIndex] < 0) {
        // We have met number = n previously
        result.push_back(n);
      } else {
        nums[flagIndex] = -nums[flagIndex];
      }
    }
    return result;
  }
};

File: algorithms/leetcode/442.find-all-duplicates-in-an-array/findDuplicates.php
<?php

// https://leetcode.com/problems/find-all-duplicates-in-an-array/
class Solution {
    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function findDuplicates($nums) {
        $result = [];
        for ($i = 0; $i < count($nums); $i++) {
            $n = $nums[$i];
            if ($n < 0) $n *= -1;
            $flagIndex = $n - 1;

            if ($nums[$flagIndex] < 0) {
                // We have met number = n previously
                $result[] = $n;
            } else {
                $nums[$flagIndex] = -$nums[$flagIndex];
            }
        }
        return $result;
    }
}